pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -168,16 +168,33 @@ func (pl *VolumeBinding) Filter(ctx context.Context, cs *framework.CycleState, p
 		}
 		return status
 	}
+
+	state.podVolumesByNodeMutex.Lock()
+	state.podVolumesByNode[node.Name] = podVolumes
+	state.podVolumesByNodeMutex.Unlock()
+	cs.Write(stateKey, state)
```

Comment:
`cs.Write` is not goroutine safe. We have to aquire `cs.Lock()`.
https://github.com/kubernetes/kubernetes/blob/f38ce5725abf09541b472004962f57fb9b42ac44/pkg/scheduler/framework/v1alpha1/cycle_state.go#L98-L103

Reply:
Thanks! fixed

Reply:
as we are saving the same pointer again, actually this is a no-op and can be omitted
I keep it as it is for clear code

Reply:
cc @ahg-g what's the best practice if the value of state key (the pointer to the actual data) does not change in the whole life?

Reply:
You don't need to call ```cs.Write``` here since the key already exist in CycleState. But you do need the lock to synchronize access to ```state.podVolumesByNode[node.Name] = podVolumes```.

Reply:
Thanks! updated